### PR TITLE
(PC-21480) feat(fetchOffers): use fetchOffers adaptors

### PR DESCRIPTION
**Pull request by EveJulliard**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-21480

Contient également les commits liés à la PR : https://github.com/pass-culture/pass-culture-app-native/pull/4677

## Checklist

I have:

- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

**delete** *if no UI change*

| Platform         | Before | After |
| :--------------- | :----: | :---: |
| iOS              |        |       |
| Android          |        |       |
| Phone - Chrome   |        |       |
| Tablet - Chrome  |        |       |
| Desktop - Chrome |        |       |
| Desktop - Safari |        |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/libs/algolia/__mocks__/mockedAlgoliaHitResponse.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { SearchResponse } from '@algolia/client-search'
 
 import { SubcategoryIdEnum } from 'api/gen'
-import { Offer } from 'shared/offer/types'
+import { AlgoliaHit } from 'libs/algolia'
 
-export const mockedAlgoliaHitsResponse: SearchResponse<Offer> = {
+export const mockedAlgoliaHitsResponse: SearchResponse<AlgoliaHit> = {
```
Comment (anoukhello) [inline src/libs/algolia/__mocks__/mockedAlgoliaHitResponse.ts:6]: En fait j'aurais bien nommé le type `AlgoliaHit` en `AlgoliaOfferHit`, t'en penses quoi ? 😬

File: src/features/search/components/Hit/Hit.native.test.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import * as logClickOnProductAPI from 'libs/algolia/analytics/logClickOnOffer'
+import { OffersWithPageFixture } from 'libs/algolia/fetchAlgolia/fetchOffers/fixtures/offersWithPageFixture'
 import { analytics } from 'libs/firebase/analytics'
 import { fireEvent, render, screen } from 'tests/utils'
 
 import { Hit } from './Hit'
 
-const mockHit = mockedAlgoliaHitsResponse.hits[0]
-const offerId = Number(mockHit.objectID)
+const mockOffer = OffersWithPageFixture.offers[0]
```
Comment (anoukhello) [inline src/features/search/components/Hit/Hit.native.test.tsx:11]: `OffersWithPageFixture` => `offersWithPageFixture ` ?

File: src/libs/algolia/fetchAlgolia/fetchOffers/fetchOffers.ts [hunk 3]
```diff
@@ -26,27 +24,36 @@ export const fetchOffers = async ({
...
+      userData: response.userData,
+    })
+
+    return adaptedOffersWithPage
   } catch (error) {
     captureAlgoliaError(error)
-    return { hits: [] as Hit<Offer>[], nbHits: 0, page: 0, nbPages: 0 }
+    return { offers: [] as Offer[], nbOffers: 0, page: 0, nbPages: 0 }
   }
 }
```
Comment (anoukhello) [inline src/libs/algolia/fetchAlgolia/fetchOffers/fetchOffers.ts]: teeeeeeeellement plus clean 😍 🔥

**Review by anoukhello — APPROVED** [review]
TROP BIEN 🔥

File: src/libs/algolia/__mocks__/mockedAlgoliaHitResponse.ts (hunk 1 above)
Comment (EveJulliard) [reply src/libs/algolia/__mocks__/mockedAlgoliaHitResponse.ts:6]: Ouai c'est mieux ! je fais ça

**Pull request merged**